### PR TITLE
DAOS-13648 test: Add support for 2.4 branches.

### PR DESCRIPTION
**Pull request by phender**
Enable generic -next_version assignment for release and weekly testing branches.

### CODE REVIEW

File: vars/daosLatestVersion.groovy [hunk 1]
```diff
@@ -23,13 +23,12 @@ String call(String next_version='1000', String repo_type='stable') {
...
-                    break
-                case 'release/2.0':
-                case 'weekly-testing-2.0':
-                    _next_version = 2.1
+                case ~/^(release|weekly).*(\/|-)(\d+\.\d+)/:
+                    version = Matcher.lastMatcher.group(3)
+                    major = version.split('.')[0] as Integer
+                    minor = version.split('.')[1] as Integer
+                    minor++
+                    _next_version = new BigDecimal("${major}.${minor}")
```
Comment (brianjmurrell) [inline vars/daosLatestVersion.groovy]: How does this all know when the _next version_ is actually a major version bump and not just a minor?

File: vars/daosLatestVersion.groovy [hunk 2]
```diff
@@ -23,13 +23,12 @@ String call(String next_version='1000', String repo_type='stable') {
...
-                case 'weekly-testing-2.2':
-                    _next_version = 2.3
-                    break
-                case 'release/2.0':
-                case 'weekly-testing-2.0':
-                    _next_version = 2.1
+                case ~/^(release|weekly).*(\/|-)(\d+\.\d+)/:
+                    version = Matcher.lastMatcher.group(3)
+                    major = version.split('.')[0] as Integer
+                    minor = version.split('.')[1] as Integer
```
Comment (brianjmurrell) [inline vars/daosLatestVersion.groovy]: Why not use regex for all of the lifting?

```suggestion
                case ~/^(release|weekly).*(\/|-)(\d+)\.(\d+)/:
                    version = 
                    major =  Matcher.lastMatcher.group(3) as Integer
                    minor =  Matcher.lastMatcher.group(4) as Integer
```

File: vars/daosLatestVersion.groovy (hunk 1 above)
Comment (phender) [reply vars/daosLatestVersion.groovy]: We can always add cases prior to this one for specific transitions and then default the minor version +1 case for anything else.

File: vars/daosLatestVersion.groovy (hunk 2 above)
Comment (phender) [reply vars/daosLatestVersion.groovy]: Seems reasonable.

**Review by brianjmurrell — DISMISSED** [review]

File: vars/daosLatestVersion.groovy (hunk 1 above)
Comment (phender) [reply vars/daosLatestVersion.groovy]: I suspect when we create `release/3.0` the `next_version` for the last `release/2.x` will still be `2.(x+1)` and it'll be `3.1` for the latest release branch.

**Review by brianjmurrell — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Pull request merged**